### DISCUSSION TITLE
Help geting command runner to execute a command on upload

### DISCUSSION THREAD

**Discussion by Sark2338** [filebrowser/filebrowser#3067] [Q&A]
I've got File Browser running just on my local network, the idea being to have a quick way to document projects and things with photos and have them all stored on my network and organized into directories, rather than on my phone in a single huge camera roll. This works great - I can take pictures with my phone and upload them via File Browser. Unfortunately, when you use the "Take Photo or Video" function of upload file, the resulting picture is always called image.jpg. The problem then comes when I want to take more than one picture in a given directory, and have to manually rename each one. To fix this, I have been trying to use the command runner function to run a simple bash script I wrote to incrementally rename the file.:
```
#!/bin/bash
FILE="$1"
numimg=`ls *.jp*g |wc -l`
name="${FILE%.*}"
ext="${FILE##*.}"
newname="$name-$numimg.$ext"
mv $FILE $newname
```
I have this saved as autorename.sh and then set command runner to execute 
```
/path/to/script/autorename.sh $FILE
```
But it doesn't work. I've tried enabling bash -c in Execute on Shell, I've tried it without it, I've tried giving my user specific permission to run all the commands and the script itself in the user permissions. I can run the script manually from the shell within File Browser and it works. But it never runs on upload.

Thinking that perhaps the problem was with running a bash script from command runner, I condensed the whole thing down into one line like so: 
```
/bin/bash -c 'numimg=`ls *.jp*g |wc -l`; name="${FILE%.*}"; ext="${FILE##*.}"; newname="$name-$numimg.$ext"; mv $FILE $newname'
```

This also does not work. Tried both with and without the Execute on Shell option set.

At this point I am stumped. I am obviously missing something but I don't know what it is. Any help would be greatly appreciated. Thanks!

**Comment by gletiecq** [comment]
I can see how this would be frustrating.  No surprise to me that filebrowser would have issues with attempts to prevent name collisions on uploads, as it's not designed to deal with that problem nor would it be happy with playing games with the file upload name on an upload.

I wonder if a web service in-between the user and fb could manage this.  User uploads to the web service, the web service manages the name issues by checking fb to see if there's a conflict, and then the web service does the upload.  That, or it doesn't think that all image uploads should be 'image.jpg' and actually uses the image name from the phone!  Not the simplest solution in the world, but I suppose it's one of the simpler options to execute.

**Reply by Sark2338** [reply]
Well, and even if it assumes all image uploads like this would be the "image.jpg" file name - this would be fine... if it would be possible to run a script automatically immediately after an upload. The script I wrote should and does work to rename the file - I actually wrote a much more involved one that checks for more possible collisions and whatnot - but I kept simplifying it further and further when I could not make it work - only to find that filebrowser simply doesn't seem to run the script at all after the file is uploaded.

Like, in theory, the functionality exists - a script to run on upload. The file name/path is, in theory, a variable provided by this process. It just doesn't work. I don't know what I'm doing wrong.

**Reply by gletiecq** [reply]
Hmm, if it's not running at all perhaps this is a permissions problem?  If fb is running as www-data or nobody that would be likely.

My daemon scripts (I recommend this practice) log filebrowser to /var/log/filebrowser.log which is pretty good at giving me diagnostics in a case like this.  Those kinds of problems were pretty easy to diagnose when I had that to work with.
